app/addons/dataimporter/stores.js

```diff
@@ -0,0 +1,369 @@
+      this.repeatTime();
+    },
+
+    getTotalRows: function () {
+      return this._totalRows;
+    },
+
+    getPreviewView: function () {
+      return this._showView;
+    },
+
+    setPreviewView: function (type) {
+      this._showView = type;
+    },
+
+    calcSmallPreviewOfData: function () {
+      //this is incase the file has large rows
+      var filesize = this._theFile.size,
+          rows = this._totalRows,
+          sizeOfEachRow = filesize / rows, //this is approximate
+          sizeCap = 800000,  //in bytes
+          numberOfRowsToShow;
+
+      numberOfRowsToShow = Math.ceil(sizeCap / sizeOfEachRow);
+      numberOfRowsToShow = numberOfRowsToShow < 500 ? 500 : numberOfRowsToShow;
+
+      this._rowsShown = numberOfRowsToShow;
+      this._smallData = this._theData.slice(0, this._rowsShown);
+    },
+
+    getSmallPreviewOfData: function () {
+      return this._smallData;
+    },
+
+    getRowsShown: function () {
+      return this._rowsShown;
+    },
+
+    getTheMetadata: function () {
+      return this._theMetadata;
+    },
+
+    getTheData: function () {
+      return this._theData;
+    },
+
+    papaparse: function () {
+      //for some reason this._config.complete gets overwritten on setconfig
+      this._config.complete = this._completeFn;
+      this._config.error = this._errorFn;
+      Papa.parse(this._theFile, this._config);
+    },
+
+    loadingComplete: function (results) {
+      this.loadMeta(results.meta);
+      this.loadData(results.data);
+      this._totalRows = this._theData.length;
+
+      if (this.isThisABigFile()) {
+        this.calcSmallPreviewOfData();
+      }
+
+      this.chunkData();
+      this.dataLoaded();
+      this.triggerChange();
+    },
+
+    chunkData: function () {
+      for (var i = 0; i < this._theData.length; i += 500) {
+        var oneChunk = this._theData.slice(i, i + 500);
+        this._chunkedData.push(oneChunk);
+      }
+    },
+
+    clearData: function () {
+      this._theData = [];
+    },
+
+    setParseConfig: function (key, value) {
+      this._config[key] = value;
+    },
+
+    getConfigSetting: function (key) {
+      return this._config[key];
+    },
+
+    getDefaultConfig: function () {
+      return {
+        delimiter : '',  // auto-detect
+        newline: '',  // auto-detect
+        header: true,
+        dynamicTyping: true,
+        preview: 0,
+        encoding: '',
+        worker: true, //true = page doesn't lock up
+        comments: false,
+        complete: function (results) {
+          this.loadingComplete(results);
+          this._changingConfig = false;
+        }.bind(this),
+        error: function () {
+          var msg1 = 'There was an error while parsing the file.',
+              msg2 = 'Please try again.';
+
+          this.goToErrorScreen('', [msg1, msg2]);
+        }.bind(this),
+        download: false,
+        skipEmptyLines: false,
+        chunk: undefined, //define function for streaming
```

Comment:
http://papaparse.com/docs#config 

it shows all of the options for papaparse. I left them in from the default so we could always see what options are available.

Should I take them out?